server/tests/rooms/game/index.test.ts

```diff
@@ -199,4 +201,18 @@ describe('inverting pending inventory', () => {
     expect(g.players['Curator'].pendingInvestments.culture).toBe(-3);
     expect(g.players['Curator'].inventory.culture).toBe(3);
   });
-});
\ No newline at end of file
+});
+
+describe('out of commission event', () => {
+  const g = new GameState(mockGameStateInitOpts(x => x, () => 10));
+  const outOfCommissionEntreprenuer = getAllMarsEvents()[7][0];
+  g.marsEvents.push(new MarsEvent(outOfCommissionEntreprenuer));
+
+
+  it('gives the entrepreneur 3 timeblocks', () => {
+    expect(g.players['Entrepreneur'].timeBlocks).toBe(10);
+    g.currentEvent.state.finalize(g);
```

Comment:
The amount of indirection going on here give me the heebie jeebies (paging @cpritcha). Sorry I didn't take the time to do a proper code review on this when it was being built. Is there a more direct way we can express this? I thought the MarsEvents could impact the game state by simply invoking `event.apply(GameState)`? Why do we have this additional finalize thing?

```
export class FinalizedMarsEvent extends KindOnlyGameEvent {
  apply(game: GameState): void {
    game.currentEvent.state.finalize(game);
  }
}
```

Reply:
`apply` will work. so create the event on its own and just call `event.apply(g)`

Reply:
cool, so are we still using `FinalizedMarsEvent` then?

Reply:
Yes. There are some MarsEvents that need to be applied at the phase (example: voting for personal gain - we only apply the effects of their voting at the end of the phase).

Reply:
So that means you will have have to either have to

- create a finalized event and add the mars event to the event list

or

- directly call `finalize` on the MarsEvent